test/test_signal_mod.py

```diff
@@ -0,0 +1,144 @@
+    test_unit = 'RPM'
+    test_minvalue = 0
+    test_maxvalue = 65535
+    test_bitsize = 16
+    test_is_signed = False
+    test_is_float = False
+
+    signal = Signal()
+    signal.set_params(test_name, test_id, test_datatype, test_init_value,
+                      test_factor, test_offset, test_unit, test_minvalue, test_maxvalue,
+                      test_bitsize, test_is_signed, test_is_float)
+
+    assert signal.name == test_name
+    assert signal.id == test_id
+    assert signal.type == test_datatype
+    assert signal.init_value == test_init_value
+    assert signal.factor == test_factor
+    assert signal.offset == test_offset
+    assert signal.unit == test_unit
+    assert signal.minvalue == test_minvalue
+    assert signal.maxvalue == test_maxvalue
+    assert signal.bitsize == test_bitsize
+    assert signal.signed == test_is_signed
+    assert signal.float == test_is_float
+
+
+def test_print_callback():
+    test_name = "name_test"
+    test_string = (
+        "#ifndef name_test_callback\n"
+        "#        define name_test_callback NULL\n"
+        "#endif"
+    )
+
+    signal = Signal()
+    signal.name = test_name
+    str_result = signal.print_callback()
+    assert str_result == test_string
+
+
+def test_print_enum():
+    test_name = "name_test"
+    test_string = "        name_test"
+
+    signal = Signal()
+    signal.name = test_name
+    str_result = signal.print_enum()
+    assert str_result == test_string
+
+
+def test_print_para_macro():
+    signal = Signal()
+    signal.name = "test_name"
+
+    for signal_type, init_value in {"U8": 255, "U16": 65000, "U32": 101, "F": 42.65}.items():
+        signal.type = signal_type
+        signal.init_value = init_value
+
+        last = False
+        test_string_not_last = f"        CAN_PARA_MACRO(test_name, sizeof({signal_type}), " \
+                               f"{init_value}, test_name_callback)"
+        str_result_not_last = signal.print_para_macro(last)
+
+        last = True
+        test_string_last = (
+            f"        CAN_PARA_MACRO(test_name, sizeof({signal_type}), "
+            f"{init_value}, test_name_callback),\n"
+            f"\n"
+            f"#ifdef E92_USE_CUSTOM_MEMSTRUCT\n"
+            f"/*******************************************************************/\n"
+            f"/*                           CUSTOM CAN                            */\n"
+            f"/*******************************************************************/\n"
+            f"        CUSTOM_CAN_SIG,\n"
+            f"#endif"
+        )
+        str_result_last = signal.print_para_macro(last)
+
+        assert str_result_not_last == test_string_not_last
+        assert str_result_last == test_string_last
+
+
+def test_print_definition():
+    signal = Signal()
+    signal.name = "test_name"
+
+    for last in [False, True]:
+        for little_endian in [True, False]:  # [LITTLE, BIG]
+            for byte_pos in [0, 8, 16, 24, 32, 40, 48, 56]:
```

Comment:
```suggestion
    last_to_test = [False, True]
    little_endian_to_test = [True, False]
    byte_pos_to_test = [0, 8, 16, 24, 32, 40, 48, 56]

    values_to_test = itertools.product(last_to_test, little_endian_to_test, byte_pos_to_test)

    for is_last, little_endian, byte_pos in values_to_test:
```

Nit-pick: Pour éviter les indentations (Et rendre le code plus lisible, **selon moi**), on peut générer une seule liste avec toutes les combinaisons à tester et itérer sur cette liste. Il faudrait ajouter `import itertools` au d/but du fichier. `itertools` est un module standard python.

Voir [itertools.product()](https://docs.python.org/3.8/library/itertools.html#itertools.product)